omas/tests/machine_mappings/test_d3d.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+import omas
+import os
+import numpy as np
+from omas.machine_mappings.d3d import ec_launcher_active_hardware
+from omas.omas_machine import tunnel_mds
+
+class TestD3D(unittest.TestCase):
+
+    def test_d3d_ECH_machine_mapping(self):
+        '''
+        Test the machine mapping of the DIII-D ECH against the a .json file created with fetchGyrotron from the OMFIT TORAY module
+        '''
+        pristine_ods = omas.load_omas_json(os.path.join('tests', 'test_data', 'd3d_ECRH_machine_mapping.json'))
+        test_ods = omas.ODS()
+        tunnel_mds('d3d', 'RF')
```

Comment:
not sure why this is necessary

Reply:
The tunnel was necessary to call the machine mapping function directly. If I used the ods.open() statement my debugger cannot find d3d.py. I will remove it now and use ods.open instead.

Reply:
Removing it produces the following error:
This produces the following, unrelated error
```
Traceback (most recent call last):
  File "/mnt/c/Users/Severin/git/omas/omas/tests/machine_mappings/test_d3d.py", line 17, in test_d3d_ECH_machine_mapping
    test_ods.load('d3d', 170325)
  File "/mnt/c/Users/Severin/git/omas/omas/omas_core.py", line 2081, in load
    results = eval('load_omas_' + ext)(*args, **kw)
  File "/mnt/c/Users/Severin/git/omas/omas/omas_machine.py", line 950, in load_omas_machine
    machine_to_omas(ods, machine, pulse, location, options, branch)
  File "/mnt/c/Users/Severin/git/omas/omas/omas_machine.py", line 207, in machine_to_omas
    f"Experiment data {data.shape} does not fit in `{location}` [{', '.join([':'] * location.count(':') + mapped.get('COORDINATES', []))}]"
Exception: Experiment data (1, 1, 331) does not fit in `equilibrium.time_slice.:.profiles_2d.:.grid_type.index` [:, :]
```
This is very weird because I can run ods.load('d3d', 170325) in other scripts. I don't know how to fix this. I submitted a commit of my current head for debugging purposes.

Reply:
Ok, I was using `omas.load()` instead of `omas.open()`. So this is fixed, but now I get this interaction:
If I do:
```
pristine_ods = omas.ODS()
pr = load_omas_json(os.path.join('omas','samples', 'd3d_ECRH_machine_mapping.json'))['ec_launchers']
pristine_ods['ec_launchers'].update(pr)
```
or
```
pristine_ods = load_omas_json(os.path.join('omas','samples', 'd3d_ECRH_machine_mapping.json'))
```
I can use `pristine_ods['ec_launchers.launcher[0]'],time()` to get the time base.
However when I do:
```
pristine_ods = omas.ODS()
pristine_ods = pristine_ods.sample()
```
with
```
@add_to_ODS
def ec_launchers(ods):
    from omas import load_omas_json

    pr = load_omas_json(os.path.join('omas','samples', 'd3d_ECRH_machine_mapping.json'))['ec_launchers']
    ods['ec_launchers'].update(pr)
    return ods
```
I can use `pristine_ods['ec_launchers.launcher[0]'],time()` returns `None`.